### ISSUE TITLE
Route rendering

### ISSUE THREAD

**Issue by andrixnet** [cgeo/cgeo#6280]
With the newly release c:geo in 2017, routing now works. 
Nearby caches are routed to, instead of straight line, if you have BRouter service installed. 

There still is no way to determine in c:geo UI if BRouter is present, properly configured and usable. Like some sort of display somewhere that an optional plugin/library is available. Not even in About / System information.
The only way (apparently) to determine this is if you request a route and "it just works". 

Using original mapsforge library (might be a limitation of this library), route segments have rectangular cut ends, which results sometimes in a jagged route display. 

If mapsforge beta is used for displaying, then route segments use rounded caps, which gives a much better display result. 

There is no way to change the color and thickness of the route line.
The red interferes heavily with default OpenStreetMap offline map default theme.

**Comment by Lineflyer** [comment]
> The only way (apparently) to determine this is if you request a route and "it just works".

Yes, this is kind of a minimum implementation right now. I am not sure if one would be able to see it by the fact, that the menu entry "Routing" is shown on the map (to select: straight, bike, etc.)

> Using original mapsforge library (might be a limitation of this library), route segments have rectangular cut ends, which results sometimes in a jagged route display.
If mapsforge beta is used for displaying, then route segments use rounded caps, which gives a much better display result.

I can confirm. I think its not worth to improve the old mapsforge library at the current point of time, as we plan to replace it by the beta anyhow.

> There is no way to change the color and thickness of the route line.
The red interferes heavily with default OpenStreetMap offline map default theme.

I dont know if it is worth to add an option here. Maybe another default color more suitable?

**Comment by andrixnet** [comment]
Might I suggest the following: 

1. an option in map settings or system settings that is 
- enabled and checked by default "Use routing service by BRouter", user can uncheck
- disabled and unchecked if BRouter service is not detected as available.

2. in map settings an option to select the following: 
route line color. 
route line thickness.
It is hard as is to select a common default, though I might suggest a blue line (personal experience with Pathaway software and routing on OSM data).

**Comment by Lineflyer** [comment]
> in map settings an option to select the following:
route line color.
route line thickness.
It is hard as is to select a common default, though I might suggest a blue line (personal experience with Pathaway software and routing on OSM data).

Just for documentation. This part of the issue has been recently solved.
